### DISCUSSION TITLE
Writing Polyglot Notebooks to quantify raw-data with MS-DIAL5 and then connect the output to MetaboAnalyst

### DISCUSSION THREAD

**Discussion by kozo2** [systemsomicslab/MsdialWorkbench#304] [Ideas]
Proposed mentors: @kozo2 @YukiMatsuzawa 

Language/skills: C# (Polyglot Notebooks), R (MetaboAnalystR), Peak picking or peak annotation using MS-DIAL

Estimated Project Length: 175 hours

Difficulty: medium

[Polyglot Notebooks](https://marketplace.visualstudio.com/items?itemName=ms-dotnettools.dotnet-interactive-vscode) is a [VScode](https://code.visualstudio.com/) extension that enables the execution of C# in [Jupyter Notebook](https://jupyter.org/). [MS-DIAL5](https://github.com/systemsomicslab/MsdialWorkbench) is written in C# and mainly used through its GUI, which is only available on Windows. However, for Linux users, MS-DIAL provides a command-line tool. This idea involves calling the functionalities of MS-DIAL's command-line tool from Polyglot Notebooks and documenting the workflow to connect the analysis results to MetaboAnalyst in a notebook. This will allow many people to learn about the workflow and reproduce it.

[MetaboAnalyst](https://www.metaboanalyst.ca/) is a comprehensive software platform designed for metabolomics data analysis and interpretation. It provides a user-friendly, web-based interface that allows researchers to perform a wide range of analyses, from basic statistical evaluations to advanced biomarker and pathway exploration. MetaboAnalyst has an accompanying [MetaboAnalystR](https://www.metaboanalyst.ca/docs/RTutorial.xhtml) R package. It automates MetaboAnalyst workflows in the R language.

In this idea, the steps for MS-DIAL5 would be documented in a C# notebook, while the steps for MetaboAnalyst would be written in an R notebook.

**Comment by Soumya624** [comment]
Hi @kozo2 & @YukiMatsuzawa ,

This seems interesting. I have a proficient background in C# and R because of my last GSoC venture at GemoScale. Would love to contribute on this. Should I solve any specific issue beforehand? What do you suggest? I have tested your platform with the .wiff data you provided earlier.

Thanks!

**Reply by kozo2** [reply]
Hi @Soumya624 .
Thank you for contacting us.
We appreciate your prior communication, but since we still do not know whether our organization application will be accepted by Google Summer of Code 2024.
Could you wait (to take any actions) until the announcement on February 21st?
https://summerofcode.withgoogle.com/
